editor/src/components/canvas/ui-jsx-canvas-renderer/ui-jsx-canvas-element-renderer-utils.tsx

```diff
@@ -84,11 +85,15 @@ export function createLookupRender(
   code: string,
   highlightBounds: HighlightBoundsForUids | null,
   editedText: ElementPath | null,
+  renderLimit: number | null,
 ): (element: JSXElement, scope: MapLike<any>) => React.ReactChild {
   let index = 0
 
   return (element: JSXElement, scope: MapLike<any>): React.ReactChild => {
     index++
+    if (renderLimit != null && index > renderLimit) {
+      return ''
```

Comment:
have you tried return null here? just commenting without actually trying this in csb but I believe null is safer